pkg/api/handler/default.go

```diff
@@ -295,184 +305,95 @@ func getTransactionsOptions(params api.POSTTransactionsParams, rejectedCallbackU
-		statusCode, arcError := m.handleError(ctx, transaction, err)
-		m.logger.Error("failed to submit transaction", slog.String("id", transaction.TxID()), slog.Int("status", int(statusCode)), slog.String("err", err.Error()))
-
-		return nil, nil, arcError
-	}
-
-	return transaction, &api.TransactionResponse{
-		Status:      int(api.StatusOK),
-		Title:       "OK",
-		BlockHash:   &tx.BlockHash,
-		BlockHeight: &tx.BlockHeight,
-		TxStatus:    tx.Status,
-		ExtraInfo:   &tx.ExtraInfo,
-		Timestamp:   m.now(),
-		Txid:        transaction.TxID(),
-		MerklePath:  &tx.MerklePath,
-	}, nil
-}
-
-func (m ArcDefaultHandler) processBEEFTransaction(ctx context.Context, transactionHex []byte, transactionOptions *metamorph.TransactionOptions) (*bt.Tx, *api.TransactionResponse, *api.ErrorFields) {
-
-	beefTx, _, err := beef.DecodeBEEF(transactionHex)
-	if err != nil {
-		errStr := fmt.Sprintf("error decoding BEEF: %s", err.Error())
-		return nil, nil, api.NewErrorFields(api.ErrStatusMalformed, errStr)
-	}
-
-	v := beefValidator.New(m.NodePolicy)
-	if err := m.validateBEEFTransaction(ctx, v, beefTx, transactionOptions); err != nil {
-		return nil, nil, err
-	}
-
-	transactions := make([]*bt.Tx, 0)
-
-	for _, tx := range beefTx.Transactions {
-		if !tx.IsMined() {
-			transactions = append(transactions, tx.Transaction)
-		}
-	}
-
-	if len(transactions) == 0 {
-		return nil, nil, api.NewErrorFields(api.ErrStatusBadRequest, "all transactions in BEEF are mined")
-	}
-
-	txStatuses, err := m.TransactionHandler.SubmitTransactions(ctx, transactions, transactionOptions)
-	if err != nil {
-		statusCode, arcError := m.handleError(ctx, nil, err)
-		m.logger.Error("failed to submit transactions", slog.Int("txs", len(transactions)), slog.Int("id", int(statusCode)), slog.String("err", err.Error()))
-		return nil, nil, arcError
-	}
-
-	lastStatus := findStatusByTxID(beefTx.GetLatestTx().TxID(), txStatuses)
-	if lastStatus == nil {
-		return nil, nil, api.NewErrorFields(api.ErrStatusNotFound, "last tx of BEEF not found in metamorph submit response")
-	}
-
-	return beefTx.GetLatestTx(), &api.TransactionResponse{
-		Status:      int(api.StatusOK),
-		Title:       "OK",
-		BlockHash:   &lastStatus.BlockHash,
-		BlockHeight: &lastStatus.BlockHeight,
-		TxStatus:    lastStatus.Status,
-		ExtraInfo:   &lastStatus.ExtraInfo,
-		Timestamp:   m.now(),
-		Txid:        lastStatus.TxID,
-		MerklePath:  &lastStatus.MerklePath,
-	}, nil
-}
-
 // processTransactions validates all the transactions in the array and submits to metamorph for processing.
-func (m ArcDefaultHandler) processTransactions(ctx context.Context, transactionsHexes []byte, transactionOptions *metamorph.TransactionOptions) ([]*bt.Tx, []interface{}, *api.ErrorFields) {
+func (m ArcDefaultHandler) processTransactions(ctx context.Context, txsHex []byte, transactionOptions *metamorph.TransactionOptions) (
+	submitedTxs []*bt.Tx, outputs []*api.TransactionResponse, failOutputs []*api.ErrorFields, pErr *api.ErrorFields) {
```

Comment:
What do you think about:
`pErr` -> `processingErr` (since we're using named returns here, we might as well make the name meaningful)
and
`outputs` -> `txResponses` / `successResponses`
`failOutputs` -> `failResponses`
to avoid confusion with transactions outputs?

And, if you agree, would be nice to also change these names in `POSTTransaction()` and `POSTTransactions()`

Reply:
I hate long names, however I changed names. Take a look.

Reply:
I like it, thanks